lib/elixir_bench/benchmarks/measurement.ex

```diff
@@ -40,10 +39,23 @@ defmodule ElixirBench.Benchmarks.Measurement do
     :percentiles
   ]
 
+  @required_fields @fields -- [:mode]
+
   @doc false
   def changeset(%Measurement{} = measurement, attrs) do
     measurement
     |> cast(attrs, @fields)
-    |> validate_required(@fields)
+    |> validate_required(@required_fields)
+    |> set_mode(attrs)
+  end
+
+  defp set_mode(changeset, attrs) do
+    mode = get_in(attrs, ["mode"])
+
+    if is_integer(mode) or is_float(mode) do
+      change(changeset, mode: [mode])
+    else
+      changeset
+    end
```

Comment:
we're not testing this functionality anywhere, are we?

Reply:
tests added